internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -419,26 +418,22 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 	return ctrl.Result{}, err
 }
 
-func (r *PreprovisioningImageReconciler) getIronicAgentImageByRelease(ctx context.Context, log logrus.FieldLogger, infraEnv common.InfraEnv) (string, error) {
-	cluster, err := r.Installer.GetClusterInternal(ctx, installer.V2GetClusterParams{ClusterID: infraEnv.ClusterID})
-	if err != nil {
+func (r *PreprovisioningImageReconciler) getIronicAgentImageByRelease(ctx context.Context, log logrus.FieldLogger, infraEnv *common.InfraEnv) (string, error) {
+	cv := &configv1.ClusterVersion{}
+	if err := r.Get(ctx, types.NamespacedName{Name: "version"}, cv); err != nil {
```

Comment:
Maybe it's better to do this once at startup and provide the image to the controller when we create it :thinking:

Reply:
It's not like the hub is going to change images after our pod starts